examples/ota-requestor-app/ota-requestor-common/BDXDownloader.cpp

```diff
@@ -69,16 +75,29 @@ void BdxDownloader::HandleTransferSessionOutput(TransferSession::OutputEvent & e
         VerifyOrReturn(err == CHIP_NO_ERROR, ChipLogError(BDX, "SendMessage failed: %" CHIP_ERROR_FORMAT, err.Format()));
         break;
     }
-    case TransferSession::OutputEventType::kAcceptReceived:
+    case TransferSession::OutputEventType::kAcceptReceived: {
+        OTAImageProcessorParams params = {
+            .imageFile = outFilePath,
+            .downloadedBytes = 0,
+            // TODO: Set this to Length field of ReceiveAccept
+            .totalFileBytes = 0,
+        };
+        mImageProcessor->SetOTAImageProcessorParams(params);
+
+        err = mImageProcessor->PrepareDownload();
+        VerifyOrReturn(err == CHIP_NO_ERROR, ChipLogError(BDX, "PrepareDownload failed: %" CHIP_ERROR_FORMAT, err.Format()));
+
         VerifyOrReturn(CHIP_NO_ERROR == mTransfer.PrepareBlockQuery(), ChipLogError(BDX, "PrepareBlockQuery failed"));
         break;
+    }
     case TransferSession::OutputEventType::kBlockReceived: {
         ChipLogDetail(BDX, "Got block length %zu", event.blockdata.Length);
 
-        // TODO: something more elegant than appending to a local file
-        // TODO: while convenient, we should not do a synchronous block write in our example application - this is bad practice
-        std::ofstream otaFile(outFilePath, std::ifstream::out | std::ifstream::ate | std::ifstream::app);
-        otaFile.write(reinterpret_cast<const char *>(event.blockdata.Data), static_cast<std::streamsize>(event.blockdata.Length));
+        ByteSpan block(event.blockdata.Data, event.blockdata.Length);
+        err = mImageProcessor->ProcessBlock(block);
```

Comment:
Same issue as prepare: this method should be asynchronous and result notified back. If not made asynchronous, then for sure `Prepare` needs to be